### PR TITLE
Prevent BoxAPIResponse from causing a NullPointerException when using a request interceptor

### PR DESCRIPTION
**Pull request by nealwu**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @nealwu, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

**Comment by nealwu** [comment]
CLA signed

**Comment by boxcla** [comment]
Verified that @nealwu has just signed the CLA. Thanks, and we look forward to your contribution.

File: src/main/java/com/box/sdk/BoxAPIResponse.java [hunk 1]
```diff
@@ -134,19 +134,21 @@ public InputStream getBody(ProgressListener listener) {
      */
     public void disconnect() {
         try {
-            if (this.rawInputStream == null) {
+            if (this.rawInputStream == null && this.connection != null) {
```
Comment (gcurtis) [inline src/main/java/com/box/sdk/BoxAPIResponse.java]: I actually think we could just put:

```java
if (this.connection == null) {
    return;
}
```

at the top of the method. I think that makes it a bit clearer that we don't need to do anything if we don't have a connection, and it avoids needing to add additional null checks.

Comment (nealwu) [inline src/main/java/com/box/sdk/BoxAPIResponse.java]: Sure, if you think that's good then let's go with that. I just wanted to be a little cautious considering that there was also a change to `this.inputStream` underneath.

**Comment by gcurtis** [comment]
Cool, looks good. Go ahead and merge it!

**Pull request merged**